lxd/storage/backend_lxd.go

```diff
@@ -2633,7 +2633,11 @@ func (b *lxdBackend) SetInstanceQuota(inst instance.Instance, size string, vmSta
 		// this will also pass empty quota for the config filesystem volume as well, allowing a former
 		// quota to be removed from both volumes.
 		if vmStateSize == "" && size != "" {
-			vmStateSize = deviceConfig.DefaultVMBlockFilesystemSize
+			if b.driver.Info().Name == "powerflex" {
+				vmStateSize = deviceConfig.DefaultVMPowerFlexBlockFilesystemSize
```

Comment:
Same point as above, we should move this into `b.driver.Info()` and avoid leaking driver specific logic.

Reply:
Added to the Info struct.